Week1/homework/world-queries.txt

```diff
@@ -37,4 +37,166 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
++----------------------------------+
+37 rows in set (0.00 sec)
+
+4. What's the top 10 countries by SurfaceArea ?
+
+mysql> SELECT name FROM country ORDER BY SurfaceArea DESC LIMIT 10 ;
++--------------------+
+| name               |
++--------------------+
+| Russian Federation |
+| Antarctica         |
+| Canada             |
+| China              |
+| United States      |
+| Brazil             |
+| Australia          |
+| India              |
+| Argentina          |
+| Kazakstan          |
++--------------------+
+10 rows in set (0.00 sec)
+
+5. What's the top 10 most populated cities ?
+
+mysql> SELECT name,population FROM city ORDER BY population DESC LIMIT 10;
++-----------------------+------------+
+| name                  | population |
++-----------------------+------------+
+| Mumbai (Bombay)       |   10500000 |
+| Seoul                 |    9981619 |
+| Sï¿½o Paulo           |    9968485 |
+| Shanghai              |    9696300 |
+| Jakarta               |    9604900 |
+| Karachi               |    9269265 |
+| Istanbul              |    8787958 |
+| Ciudad de Mï¿½xico    |    8591309 |
+| Moscow                |    8389200 |
+| New York              |    8008278 |
++-----------------------+------------+
+10 rows in set (0.01 sec)
+
+6. Select the top 3 countries by population that their names start with the letter ‘P’
+mysql> SELECT name,population FROM country WHERE name LIKE 'p%' ORDER BY population Desc LIMIT 3 ;
++-------------+------------+
+| name        | population |
++-------------+------------+
+| Pakistan    |  156483000 |
+| Philippines |   75967000 |
+| Poland      |   38653600 |
++-------------+------------+
+3 rows in set (0.01 sec)
+
+7.What is the population in Asia?
+
+mysql> SELECT sum(population) FROM country WHERE continent='Asia';
++-----------------+
+| sum(population) |
++-----------------+
+|      3705025700 |
++-----------------+
+1 row in set (0.01 sec)
+
+8. Which languages are spoken in the Region of ‘South America'?
+
+mysql> SELECT distinct language FROM countrylanguage JOIN country ON country.code=countrylanguage.countrycode WHERE country.region='South America';
++------------------+
+| language         |
++------------------+
+| Indian Languages |
+| Italian          |
+| Spanish          |
+| Aimarï¿½         |
+| Guaranï¿½        |
+| Ketï¿½ua         |
+| German           |
+| Japanese         |
+| Portuguese       |
+| Araucan          |
+| Rapa nui         |
+| Arawakan         |
+| Caribbean        |
+| Chibcha          |
+| Creole English   |
+| English          |
+| Creole French    |
+| Hindi            |
+| Sranantonga      |
+| Goajiro          |
+| Warrau           |
++------------------+
+21 rows in set (0.00 sec)
+
+9. What are the languages spoken on all cities named ‘Barcelona’ (you may need to join 3 tables =) ) 
+
+mysql> SELECT distinct language FROM countrylanguage JOIN city ON countrylanguage.countrycode=city.countrycode WHERE city.name='Barcelona';
```

Comment:
Interesting....  `countrylanguage` and `city` are not directly related.. but they have an implicit connection through `countrycode`...  best practice is to join tables that are directly related `countrylanguage` with `country` and that result join it again with `city` (or the other way around) (Ill tell Nadine to explain this one better in class, if she doesn't please ask about this one !)

As a sidenote part of the fun of this query (besides trying to join 3 tables is also realizing there is more than one country with a city named Barcelona) so you can also return country name, city name (will be a column with all values Barcelona) and the language.. then you'll need to remove the distinct ! Give it a try !

If you need the entity-relation diagram 
http://4.bp.blogspot.com/-BUJLojNo4ik/UNIqWTKGwHI/AAAAAAAAAAY/DA4YD6EGJh4/s1600/world_erd.png